npm/angular/src/mount.ts

```diff
@@ -163,18 +172,18 @@ function initTestBed<T> (
   component: Type<T> | string,
   config: MountConfig<T>,
 ): Type<T> {
-  const { providers, ...configRest } = config
+  const { componentProviders, ...configRest } = config
 
   const componentFixture = createComponentFixture(component) as Type<T>
 
   getTestBed().configureTestingModule({
     ...bootstrapModule(componentFixture, configRest),
   })
 
-  if (providers != null) {
+  if (componentProviders != null) {
     getTestBed().overrideComponent(componentFixture, {
-      add: {
```

Comment:
Do you know the difference between `add` and `set` here? Hard to find documentation on this and I'm curious

Reply:
Yeah, so with `add`, it is kind of a merge between the existing providers and the ones that you add in the test. With `set`, you replace them.

That's why you need to use `set` when you want to be able to remove all component providers (as the last test does)

Reply:
Thinking more on this, what is the behavior when specifying `componentProviders` when using the 
```ts
cy.mount('<some-component />', { componentProviders: [SomeProvider], declarations: [SomeComponent] })
```

form? The providers would be attached to the WrapperComponent in this case, would `SomeComponent` inherit the providers?

Reply:
It will inherit it, but if `SomeComponent` has its own providers, they will not be replaced. With a wrapper component it doesn't make any difference, if I use `providers` or `componentProviders`.

Theoretically, we could spend some effort trying to figure out that `SomeComponent` is embedded, but this is not an ideal solution. Just think of the following case:

```typescript
cy.mount('<div><component-a></component-a><component-b></component-b></div>', {componentProviders: [SomeProvider]});
```

Now we have two components. To which one should we assign the `SomeProvider`?

The question is if it actually makes sense to provide `componentProviders` for a WrapperComponent. It might lead to further confusion.

Let me check, how other libraries deal with this situation and I'll come back to you.

What are your thoughts?

Reply:
I need to run through some test-cases and map out how the providers at any given level are able to be overridden. It's been a while since I've worked closely with Angular, but I'm starting to think we could cut `componentProviders` from the scope of this PR and move forward with the `providers` switch to be module-level. There might be a use-case for it in the future but I don't see the benefits of `componentProviders` clearly right now.

Reply:
In Angular testing, you have two types of how to override providers.

The first is the providers of the `TestBed.configureTestingModule`, and the second one is the one in `TestBed.overrideComponent`. You need both.

With `componentProviders` you can override the providers inside the `@Component({providers: ...})`. `providers` of the TetsingModule will not reach them.

In my test, you see the difference between these two.

Reply:
A typical use case for `componentProviders` is the @ngrx/component-store. This is a service, you usually provide inside the `@Component({})`: https://ngrx.io/guide/component-store/initialization#lazy-initialization